Jenkinsfile

```diff
@@ -77,20 +82,47 @@ node('kiali-build && fedora') {
         returnStdout: true,
         script: "sed -rn 's/^VERSION \\?= v(.*)/\\1/p' Makefile").trim().replace("-SNAPSHOT", "")
     def nextVersion = bumpVersion(params.RELEASE_TYPE, releasingVersion)
+    if (params.RELEASE_TYPE == 'patch') {
+      // If we are doing a patch release, the Makefile contains a version that is already released.
+      // The version present in the Makefile needs to be bumped and that's our releasing version.
+      // The next version is two patches from whatever is present in the Makefile
+      releasingVersion = nextVersion
+      nextVersion = bumpVersion(params.RELEASE_TYPE, releasingVersion)
+
+      // If we are not doing a patch, we assume a minor is being built. In this case,
+      // the Makefile already stores the version we want to release, so no need to do further calcs.
+    }
 
     stage('Build and release Helm charts') {
       // Build the release
       echo "Will build version: ${releasingVersion}"
-      sh "sed -i -r 's/^VERSION \\?= v.*/VERSION \\?= v${releasingVersion}/' Makefile"
-      sh "make clean update-helm-repos"
+      sh "make -e VERSION=v${releasingVersion} clean build-helm-charts"
+
+      if (params.RELEASE_TYPE == 'patch') {
+        // Switch to `master` branch before updating docs/index.yaml
+        sh "git checkout master"
+      } else {
+        // Anticipated preparation of Makefile so that it contains the released
+        // version when creating the git tag.
+        //
+        // This change to the Makefile is not done for patch releases, because that
+        // would break the next minor build (remember we have already switched to master).
+        sh "sed -i -r 's/^VERSION \\?= v.*/VERSION \\?= v${releasingVersion}/' Makefile"
+      }
+
+      sh "make -e VERSION=v${releasingVersion} update-helm-repos"
 
       // Tag the release
+      //   Note that if we are doing a patch release, this tag won't contain valid `kiali-server` nor `kiali-operator` directories
```

Comment:
I think we should no longer create tags. They don't seem to be that useful. For patch releases, the tag will only contain the updated index.yaml and the tarballs, which will already be in master.

Probably, only the vX.Y branch is enough. But I'm not sure...

Reply:
I like tags for historical purposes. We can see exactly which commit produced which patch build. So at least we have it for historical and documentation purposes.

Reply:
Note - I did not follow the code exactly, but when I say I like the tag for historical purposes, the tag I mean is the "vX.Y.Z" tag that is created on the git commit where the build-helm-repos is run.